src/device/devicemanager.cpp

```diff
@@ -221,7 +221,7 @@ void DeviceManager::ListerClosed() {
 
 void DeviceManager::DeviceDestroyed() {
 
-  ConnectedDevice *device = static_cast<ConnectedDevice*>(sender());
+  ConnectedDevice *device = dynamic_cast<ConnectedDevice*>(sender());
```

Comment:
`dynamic_cast` does not work here when `sender()` is destroyed, that's the reason `qobject_cast` wasn't used here in the first place. This will crash.